shared/src/types/key/secp256k1.rs

```diff
@@ -304,6 +310,7 @@ impl Serialize for Signature {
         for elem in &arr[..] {
```

Comment:
On L309
```
        let mut seq = serializer.serialize_tuple(arr.len())?;
```

should this be `arr.len() + 1` now that we have the extra `seq.serialize_element()` call? We should test our Serialize/Deserialize impls match up (we should already have had tests probably, I could have a go at this even as a separate PR)

Reply:
good catch! Will change and cover with tests.